dxil_converter.cpp

```diff
@@ -4340,6 +4340,15 @@ spv::StorageClass Converter::Impl::get_effective_storage_class(const llvm::Value
 		return fallback;
 }
 
+bool Converter::Impl::get_needs_temp_storage_copy(const llvm::Value *value) const
+{
+	auto itr = needs_temp_storage_copy.find(value);
```

Comment:
This needs to check if value is an access chain, basically anything that is not llvm::AllocaInst.

Reply:
I don't think this is true, consider the following:

```
blah = alloca
TraceRay(..., blah)
CallShader(..., blah)
```

We'd still need to have a temporary storage copy for CallShader and TraceRay as there is no access chain in between and they require different OpVariables but on the same alloca

Reply:
Right, but I mean `!alloca || needs_temp_storage.count(value) != 0`.

Reply:
Resolved locally.